library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -311,6 +313,10 @@ private Slice getFromReaders(Slice key, long timeBucket)
     @Override
     public Slice get(Slice key, long timeBucket, ReadSource readSource)
     {
+      if (key.getClass() == Slice.class) {
+        //The hashCode of Slice was not correct, so correct it
+        key = new SerializeSlice(key);
+      }
```

Comment:
I am also in favor of using ByteBuffer (with the ability to switch between HeapByteBuffer and DirectByteBuffer). 
I guess Slice was introduced to Netlet as ByteBuffer provides more functionality but also is more heavyweight. It is not Kryo serializable, but it is unlikely that such ability is required.